pilot/pkg/bootstrap/server.go

```diff
@@ -884,7 +885,7 @@ func (s *Server) initRegistryEventHandlers() {
 			pushReq := &model.PushRequest{
 				Full: true,
 				ConfigsUpdated: map[model.ConfigKey]struct{}{{
-					Kind:      curr.GroupVersionKind,
+					Kind:      kind.FromGvk(curr.GroupVersionKind),
```

Comment:
This is the risk area of "unknown type". I don't think this _should_ happen and the `panic` should capture any case it does -- hopefully in tests and not in users code.So seems pretty safe

Reply:
if panic is not nice I could add some `Unknown` constant and return that